src/Npgsql/Internal/NpgsqlConnector.cs

```diff
@@ -2302,6 +2302,10 @@ void PerformKeepAlive(object? state)
                     return;
 
                 Log.Trace("Performing keepalive", Id);
+                AttemptPostgresCancellation = false;
```

Comment:
Don't we need to restore this to what it was? Otherwise a keepailve could cause a normal query to break the connector instead of attempting a PG cancellation?

Also just to make sure that setting the UserTimeout below doesn't have an effect on the regular query later, in a similar way.

Reply:
Don't really have to, since timeout for a normal query (and `AttemptPostgresCancellation`) is set whenever a user action is started.

But sure, I can make a test just to confirm this.

Reply:
No, you're right - no need to add a test for this.